crates/bevy_ecs/src/bundle.rs

```diff
@@ -665,6 +672,7 @@ impl<'a, 'b> BundleSpawner<'a, 'b> {
 pub struct Bundles {
     bundle_infos: Vec<BundleInfo>,
     bundle_ids: HashMap<TypeId, BundleId>,
+    bundle_ids_dynamic: HashMap<Vec<ComponentId>, BundleId>,
```

Comment:
The `Vec<ComponentId>` could be reduced to a `Box<[ComponentId]>` since it's never appending elements, therefore the entire `Vec` is not needed.

Reply:
Creating boxed slices will induce a lot of unnecessary reallocations which are far more expensive than the 8 bytes you save of not having a `Vec`.

Reply:
ahh yes true, forgot the Vec would need to drop any excess storage and possibly reallocate. Not just a simple pointer transfer.